### ISSUE TITLE
5.0 features and roadmap

### ISSUE THREAD

**Issue by jondayton** [iunu/mobx-async-store#377]
It's been a while since we made major updates. I'd like to start collecting ideas here for critical changes.

### Essential
- [ ] Replace inferred inverse relationships for `relatedToMany` with explicit ones
- [ ] Model decorators use options instead of being limited to one constructor param
- [ ] Upgrade to latest `mobx` version (blocked by next one)
- [ ] TBD: either move away from experimental decorators, or don't use inheritance for `Model` and `Store`
- [ ] Move to Typescript

### Nice to have
- [ ] Offline support / queuing
- [ ] GraphQL serialization
